backend/data/src/main/kotlin/io/tolgee/batch/BatchJobProjectLockingManager.kt

```diff
@@ -134,12 +134,32 @@ class BatchJobProjectLockingManager(
 
   private fun getInitialJobId(projectId: Long): Long? {
     val jobs = batchJobService.getAllIncompleteJobIds(projectId)
+
+    // First priority: Find actually RUNNING jobs from database
+    // This prevents phantom locks from PENDING jobs that have chunks in queue but aren't executing
+    val runningJob = jobs.find { job ->
+      val jobDto = batchJobService.getJobDto(job.jobId)
+      jobDto.status == io.tolgee.model.batch.BatchJobStatus.RUNNING
```

Comment:
status can be easily added to the query inside the `getAllIncompleteJobIds` method. Also, it is used only here, so it can be safely updated. It will save additional trips to a database.